### PR TITLE
Start a rever file for the release

### PR DESCRIPTION
**Pull request by asmeurer**
See http://www.ergs.sc.edu/rever-docs/.

CC @certik

### CODE REVIEW

**Comment by certik** [comment]
Cool. I was hoping to use docker instead of a virtualbox to automate things.

**Comment by certik** [comment]
I see that @asmeurer independently suggested the same thing, I am copying what he said on the mailinglist:
```
I started a PR here https://github.com/sympy/sympy/pull/12819

Anthony Scopatz and I are working on a new tool to replace the current
release/fabfile.py https://github.com/ergs/rever. The existing script
uses a Vagrant VM and fabric. What I would like to replace this with
is a Docker image, that runs the rever script.

Rever is designed to be a general releasing tool that lets you write
the different release tasks as "activities" in a xonsh script. Xonsh,
if you don't know, if a shell that mixes bash and Python, so it's
basically Python but you can write shell commands anywhere that you
would have a Python statement.

If you want to help out, a good place to start would be to build a
Docker container. All the dependencies that need to be installed on it
are in the fabfile
https://github.com/sympy/sympy/blob/master/release/fabfile.py. We need
to be able to run everything in the Python versions we support (2.7,
3.3-3.6). I recommend using conda environments for everything if
possible. I have the sympy name on dockerhub, and can give you access
if you give me your username.
```

**Comment by asmeurer** [comment]
Yes, let's use docker. Can you start a container based on the stuff that's installed [here](https://github.com/sympy/sympy/blob/cd3224018333211374b2fe914705825e5125a3b1/release/fabfile.py#L130)? Also, we should use conda for the Python, instead of the packages from apt, so that we can easily test all the Python versions, and it will be easier to test in isolated environments. 

If you give me your dockerhub username, I can add you to the sympy user. 

Also, if you are interested, I can show you how the `rever` stuff works.

**Comment by asmeurer** [comment]
So the idea for replacing https://github.com/sympy/sympy/blob/master/release/fabfile.py:

- Instead of a Vagrant box we have a Dockerfile, and we upload the stuff to the `sympy` user on DockerHub.

- All the stuff currently done in the fabfile for installing packages or clearing things out will not be needed for Docker, because it already does that. 

- `rever` handles the release actions themselves. The point of using rever is that we can factor out some of the more complicated logic, like the tarball whitelist check or uploading to GitHub to the rever library itself.

- The docker file has `rever` as its entry point, and uses the release directory as a shared volume (so that the final release files can be pulled out of the container). 

Feel free to push to this branch (I have allowed edits from maintainers). Once I finish all the issues in the [1.1 milestone](https://github.com/sympy/sympy/milestone/46) I will create a `1.1` release branch on `sympy/sympy`.

**Comment by asmeurer** [comment]
Merging this so I can start the release branch.
